### DISCUSSION TITLE
F# identifiers alphabet

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/fsharp#12482] [Q&A]
I noticed that we can use greek unicode symbols in F# but not (say) APL symbols. I can write a function named ρ but cannot write a function named ⍋.

Is there a reason for this restriction on the alphabet? is it constrained by .Net and the CLR?

**Comment by dsyme** [comment]
This should go to https://github.com/fsharp/fslang-suggestions/


> I noticed that we can use greek unicode symbols in F# but not say, APL symbols. I can write a function named ρ but cannot write a function named ⍋.

I'd imagine you can use ``⍋``

I'm actually open to changing the F# spec to allow APL symbols in identifiers if there's a good clean spec of what's allowed.  There's a reasonable practical use-case for this after all (porting APL code to F#)

**Reply by The-Futurist** [reply]
@dsyme - That's encouraging, OK I spent some time today putting a suggestion together, I'll leave this to stew for a while and see how it goes.

Thanks
